lib/scale/decode.go

```diff
@@ -342,6 +342,12 @@ func (sd *Decoder) DecodeArray(t interface{}) (interface{}, error) {
 				copy(arr[:], buf)
 				*ptr = arr
 			}
+		case *common.PeerInfo:
```

Comment:
was this the one way to do this? I guess it's blocked on the "support arrays of structs" issue?

Reply:
right, I'll start working on that issue.